src/pybind/mgr/dashboard/controllers/smb.py

```diff
@@ -88,6 +89,24 @@
 }
 
 
+def raise_on_failure(func):
+    @wraps(func)
+    def wrapper(*args, **kwargs):
+        result = func(*args, **kwargs)
+
+        if isinstance(result, dict) and 'success' in result and result['success'] is False:
+            msg = 'Operation was unsuccessful'
+            if isinstance(result['results'], list):
+                msg = result['results'][0]['msg'] if 'msg' in result["results"][0] else msg
+            else:
+                msg = result['msg'] if 'msg' in result["results"] else msg
+            raise DashboardException(msg=msg, component='smb')
```

Comment:
sorry  for being dump here but can you add inline comments to this part so its more understandable.

Reply:
The idea here is to read the response of an SMB operation, if the response fails for any reason it will return a dictionary with a `msg` key that will contain a message with the reason for failure, hence we capture that with this decorator and raise a DashboardException so the error message is displayed as a notification in the Dashboard.

I can look to add a short explanation

Reply:
I've updated it, let me kniow what you think!